src/test/java/com/google/research/bleth/simulator/TracingSimulationTest.java

```diff
@@ -0,0 +1,374 @@
+package com.google.research.bleth.simulator;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TracingSimulationTest {
+
+    private static final IMovementStrategy MOVE_UP = new moveUp();
+    private static final IMovementStrategy STATIONARY = new StationaryMovementStrategy();
+    private static final AwakenessStrategyFactory DUMMY_AWAKENESS_STRATEGY_FACTORY = new AwakenessStrategyFactory();
+
+    private static final String SIMULATION_ID = "test-sim-id-1";
+    private static final int BOARD_DIMENSION_EQUALS_TWO = 2;
+    private static final int BOARD_DIMENSION_EQUALS_THREE = 3;
+    private static final int MAX_ROUNDS_EQUALS_TWO = 2;
+    private static final int NUMBER_OF_BEACONS_EQUALS_TWO = 2;
+    private static final int NUMBER_OF_OBSERVERS_EQUALS_TWO = 2;
+    private static final double RADIUS_EQUALS_ONE = 1.0;
+
+    private static final int ILLEGAL_BOARD_DIMENSION = 0;
+    private static final int ILLEGAL_NUMBER_OF_AGENTS = 0;
+    private static final int ILLEGAL_MAX_ROUNDS = 0;
+    private static final int ILLEGAL_RADIUS = -1;
```

Comment:
Nit: Consider name it NEGATIVE_RADIUS; I assume 0 is an illegal value for radius as well.